packages/devtools_app/lib/src/ui/html_elements.dart

```diff
@@ -5,8 +5,18 @@
 import 'dart:async';
 import 'package:html_shim/html.dart' hide Point;
 
+/// Override of the default document root to allow DevTools to work
+/// inside of a shadow DOM.
+///
+/// When this is not null, [querySelector] calls should use it
+/// instead of [document.querySelector(selectors)] to retrieve
+/// elements.
+Element overrideDocumentRoot;
+
 /// Finds the first descendant element of this document with the given id.
-Element queryId(String id) => querySelector('#$id');
+Element queryId(String id) => overrideDocumentRoot != null
```

Comment:
nice. I wouldn't be very confident this is the only way we are querying by id or querySelector so definitely audit the debugger code to make sure.

Reply:
There's also a call to querySelectorAll in framework, but it's for the main navbar, so it's not needed.